packages/vm/src/cache.rs

```diff
@@ -291,11 +292,22 @@ where
         // Re-compile from original Wasm bytecode
         let wasm = self.load_wasm_with_path(&cache.wasm_path, checksum)?;
         cache.stats.misses = cache.stats.misses.saturating_add(1);
-        // Module will run with a different engine, so we can set memory limit to None
-        let engine = make_compiling_engine(None);
-        let module = compile(&engine, &wasm)?;
-        // Store into the fs cache too
-        let module_size = cache.fs_cache.store(checksum, &module)?;
+        {
+            // Module will run with a different engine, so we can set memory limit to None
+            let compiling_engine = make_compiling_engine(None);
+            // This module cannot be executed directly as it was not created with the runtime engine
+            let module = compile(&compiling_engine, &wasm)?;
```

Comment:
👍🏼 Tried to compile here with the runtime engine directly, but then some tests fail with

`CompileErr { msg: "Could not compile: Compilation error: No compiler compiled into executable" }`

Reply:
Yeah once you start understanding how this stuff is supposed to work, it all starts to make sense. Being able to have a lightweight engine without compiler is pretty powerful.